geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync } from 'fs';
+const mapshaper = require('mapshaper');
+const unzipper = require('unzipper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private _unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          unzipper.Extract({
+            path: fileInfo.destination + fileInfo.filename.replace('.zip', ''),
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
+        )
+        .on('error', () =>
+          reject(new Error(`${fileInfo.filename} could not been extracted`)),
+        );
+    });
+  }
+
+  private async _shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    const outputKey = `shapefile-${new Date().getTime()}.gejson`;
+    const _geoJson = await mapshaper.applyCommands(
+      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -o ${outputKey}`,
+    );
+
+    return JSON.parse(_geoJson[outputKey].toString('utf-8'));
+  }
+
+  private wipeOutFolder(path: string): void {
+    rmdirSync(path, { recursive: true });
+  }
+
+  async getGeoJson(shapeFile: Express.Multer.File) {
```

Comment:
Strongly recommending to add return type (`Promise<Geometry>` ? not sure, please verify)

Not sure as well if shapefile validates GeoJSON as well.

Reply:
I would suggest to reconsider this approach - we can keep it for this PR, but reading a potentially large file and passing it around forces us to keep it in memory, and depending on context deal with serialization/deserialization/etc - I would consider an approach where we only pass around the path to the geojson file written for us by mapshaper on disk, and then fopen *and stream* that file's contents directly into the HTTP response.

Reply:
@hotzevzl not sure if this is possible to reach the disk. The file is created on `GeoService` module, which would introduce limitations how we deploy units (Api, Geo).  Currently this sends the whole thing over HTTP to API and back to consumer. Probably it would require putting the file on some S3-like storage with short lifespan (to be auto-removed) and provide the link to it back to consumer (or maybe even delegate streaming the file to some cloud function).

Not sure about actual limit (of file size) we have in mind but looks like that different (than current) approach would introduce a lot of things to implement now.

Reply:
@kgajowy to clarify my suggestion: I am thinking about something like this: https://github.com/mbloch/mapshaper/wiki/Using-mapshaper-programmatically#runcommands instead of `applyCommands()`, and then open the GeoJSON file created by mapshaper on disk and stream that back to the response from geoprocessing to api, which will then reverse-proxy this back to the API client.
Using a storage bucket and only returning the signed URL of a short-lived file on a bucket could also be an option, but we'd still need to get the file written to local disk first (if wanting to avoid keeping the whole GeoJSON document in memory) and then use an Azure storage client module to write it to a bucket, sign it, and so on.

In practice I am happy with the current implementation for this PR, of course - no need to add scope to it now!